examples/local/backups/restart_tablets.sh

```diff
@@ -23,19 +23,31 @@ for i in 100 101 102; do
   CELL=zone1 TABLET_UID=$i ./scripts/mysqlctl-up.sh
   CELL=zone1 KEYSPACE=commerce TABLET_UID=$i ./scripts/vttablet-up.sh
 done
-sleep 5
-vtctldclient InitShardPrimary --force commerce/0 zone1-100
 
 for i in 200 201 202; do
   CELL=zone1 TABLET_UID=$i ./scripts/mysqlctl-up.sh
   SHARD=-80 CELL=zone1 KEYSPACE=customer TABLET_UID=$i ./scripts/vttablet-up.sh
 done
-sleep 5
-vtctldclient InitShardPrimary --force customer/-80 zone1-200
 
 for i in 300 301 302; do
   CELL=zone1 TABLET_UID=$i ./scripts/mysqlctl-up.sh
   SHARD=80- CELL=zone1 KEYSPACE=customer TABLET_UID=$i ./scripts/vttablet-up.sh
 done
 sleep 5
+
+# Wait for all the replica tablets to be in the serving state before initiating
+# InitShardPrimary. This is essential, since we want the RESTORE phase to be
+# complete before we start InitShardPrimary, otherwise we end up reading the
+# tablet type to RESTORE and do not set semi-sync, which leads to the primary
+# hanging on writes.
+for i in 101 201 301; do
+  for _ in $(seq 0 300); do
+   status=$(curl "http://$hostname:15$i/debug/status_details")
+   echo "$status" | grep "REPLICA: Serving" && break
+   sleep 0.1
+  done
+done
+
+vtctldclient InitShardPrimary --force commerce/0 zone1-100
+vtctldclient InitShardPrimary --force customer/-80 zone1-200
 vtctldclient InitShardPrimary --force customer/80- zone1-300
```

Comment:
nit: can you add the trailing newline here?

Reply:
fixed